### DISCUSSION TITLE
One JS with all Plugins and Features

### DISCUSSION THREAD

**Discussion by samety22** [6pac/SlickGrid#847] [Q&A] [answered]
Hi,

i have a Question. For some Tables with additional Features i need to import a lot of JS files like this example:

```
<script src="../dist/browser/slick.core.js"></script>
<script src="../dist/browser/slick.interactions.js"></script>
<script src="../dist/browser/slick.grid.js"></script>
<script src="../dist/browser/slick.dataview.js"></script>
<script src="../dist/browser/controls/slick.columnpicker.js"></script>
<script src="../dist/browser/controls/slick.gridmenu.js"></script>
```

and for an other Table on the Same Page i need again this one:

```
<script src="../dist/browser/slick.core.js"></script>
<script src="../dist/browser/slick.interactions.js"></script>
<script src="../dist/browser/slick.grid.js"></script>
<script src="../dist/browser/plugins/slick.cellrangedecorator.js"></script>
<script src="../dist/browser/plugins/slick.cellrangeselector.js"></script>
<script src="../dist/browser/plugins/slick.cellselectionmodel.js"></script>
<script src="../dist/browser/plugins/slick.rowselectionmodel.js"></script>
<script src="../dist/browser/plugins/slick.crossgridrowmovemanager.js"></script>
<script src="../dist/browser/slick.formatters.js"></script>
<script src="../dist/browser/slick.editors.js"></script>
```

So there are alot of JS files to import. My Question: is there a ALLSLICK.JS or ALLSLICKPLUNGINS.JS ... something like that?

thanks

**Comment by 6pac** [comment]
Usually web developers use a bundling and minification tool to handle all of this stuff. You don't have bundling as part of your stack?

**Reply by samety22** [reply]
Hi, thanks for your answer. I use Vanilla JS (no react or other librarys) so i dont have a bundler. I must install a Software for that. It would be helpful if SlickGrid has its own bundle.js in the Package.

**Comment by ghiscoding** [comment, marked as answer]
Well if you use the new ESM import then it's 1 import, see [Example 4 - ESM](http://6pac.github.io/SlickGrid/examples/example4-model-esm.html) which is in vanilla JS and just requires you to use `<script type="module">`

As for the old browser (`iife`) imports, that will not change, you still need to import multiple files and we do not expect to ever change it because it is still a good way to only import what you want/use and that is what SlickGrid users have been using and loading in their pages for the past 10 years. For example, if we merged everything into 1 bundle, you would be loading the entire libraries with all controls & plugins and you must probably will never use them all in most cases. Again it's recommended to use the new ESM import (which is bundled into 1 file), 

Actually now that I think of it, you could create your own "all-in-one" bundle, you just need to move these imports into an external JS file that you can reuse and then load this custom "all-in-one" JS file into your project. That way you could customize it to your need and only include what you want.
